### PR TITLE
Add somatic to DSL2

### PR DESCRIPTION
**Pull request by MaxUlysse**
# nf-core/sarek pull request

Many thanks for contributing to nf-core/sarek!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/sarek branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/sarek)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by FriederikeHanssen** [comment]
FYI: I think the BioContainers for bwa-mem2 have been updated. It is not shown yet on the website but they are listed on https://quay.io/repository/biocontainers/bwa-mem2?tab=tags and https://depot.galaxyproject.org/singularity/ . Mulled container is still missing though as far as I see: https://quay.io/repository/biocontainers/mulled-v2-fe8faa35dbf6dc65a0f7f5d4ea12e31a79f73e40?tab=tags

**Comment by FriederikeHanssen** [comment]
> FYI: I think the BioContainers for bwa-mem2 have been updated. It is not shown yet on the website but they are listed on https://quay.io/repository/biocontainers/bwa-mem2?tab=tags and https://depot.galaxyproject.org/singularity/ . Mulled container is still missing though as far as I see: https://quay.io/repository/biocontainers/mulled-v2-fe8faa35dbf6dc65a0f7f5d4ea12e31a79f73e40?tab=tags

There is anew bwa-mem2 container now: 

 conda (params.enable_conda ? "bioconda::bwa-mem2=2.1--he513fc3_0 bioconda::samtools=1.11--h6270b1f_0" : null)
    if (workflow.containerEngine == 'singularity' && !params.pull_docker_container) {
        container "https://depot.galaxyproject.org/singularity/mulled-v2-e5d375990341c5aef3c9aff74f96f66f65375ef6:e6f0d20c9d78572ddbbf00d8767ee6ff865edd4e-0" 
    } else {
        container "quay.io/biocontainers/mulled-v2-e5d375990341c5aef3c9aff74f96f66f65375ef6:e6f0d20c9d78572ddbbf00d8767ee6ff865edd4e-0"
    }

File: main.nf [hunk 1]
```diff
@@ -94,6 +94,8 @@ anno_list = define_anno_list()
 annotate_tools = params.annotate_tools ? params.annotate_tools.split(',').collect{it.trim().toLowerCase().replaceAll('-', '')} : []
 if (!check_parameter_list(annotate_tools,anno_list)) exit 1, 'Unknown tool(s) to annotate, see --help for more information'
 
+if (!(params.aligner in ['bwa-mem', 'bwa-mem2'])) exit 1, 'Unknown aligner, see --help for more information'
```
Comment (FriederikeHanssen) [inline main.nf:97]: Should we keep both aligners or remove bwa-mem? Maybe depends on whether bwa-mem is further developed and not replaced with bwa-mem2

**Review by FriederikeHanssen — APPROVED** [review]

File: main.nf (hunk 1 above)
Comment (MaxUlysse) [reply main.nf:97]: I'd say let's keep both for now, but let's keep that question in mind.
https://github.com/nf-core/sarek/issues/337

**Comment by MaxUlysse** [comment]
> > FYI: I think the BioContainers for bwa-mem2 have been updated. It is not shown yet on the website but they are listed on https://quay.io/repository/biocontainers/bwa-mem2?tab=tags and https://depot.galaxyproject.org/singularity/ . Mulled container is still missing though as far as I see: https://quay.io/repository/biocontainers/mulled-v2-fe8faa35dbf6dc65a0f7f5d4ea12e31a79f73e40?tab=tags
> 
> There is anew bwa-mem2 container now:
> 
> ```
>  conda (params.enable_conda ? "bioconda::bwa-mem2=2.1--he513fc3_0 bioconda::samtools=1.11--h6270b1f_0" : null)
>     if (workflow.containerEngine == 'singularity' && !params.pull_docker_container) {
>         container "https://depot.galaxyproject.org/singularity/mulled-v2-e5d375990341c5aef3c9aff74f96f66f65375ef6:e6f0d20c9d78572ddbbf00d8767ee6ff865edd4e-0" 
>     } else {
>         container "quay.io/biocontainers/mulled-v2-e5d375990341c5aef3c9aff74f96f66f65375ef6:e6f0d20c9d78572ddbbf00d8767ee6ff865edd4e-0"
>     }
> ```

I noticed that samtools is on version 1.11 now, I'll try and update the bwa mem container with the samtools 1.10

**Pull request merged**